### ISSUE TITLE
False positive: `multiple-statements` (`C0321`) should not trigger for stub functions

### ISSUE THREAD

**Issue by GideonBear** [PyCQA/pylint#7860]
### Bug description

For example:
```py
class MyClass(ABC):
    @abstractmethod
    def function(self) -> int: ...  # this should not raise `multiple-statements` error as it's a stub
```
`...` is technically a statement, but shouldn't be considered one.

If accepted, I would be interested in submitting a PR for this.

### Command used

```shell
N/A
```

### Pylint output

```shell
C0321 More than one statement on a single line (multiple-statements)
```

### Expected behavior

No error

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.12.13
Python 3.9.15 (main, Nov 11 2022, 13:33:56) 
[GCC 9.4.0]
```

**Comment by mbyrnepr2** [comment]
Currently, functions decorated with `typing.overload` are permitted to have it on the same line; so there is precedent in pylint for this proposed change.

**Comment by GideonBear** [comment]
> Is there a PEP that says otherwise?

PEP 8 doesn't say it specifically, but in [other recommendations](https://peps.python.org/pep-0008/#other-recommendations) it does put `...` on the same line in some examples.

**Comment by mbyrnepr2** [comment]
Are you importing abstractmethod from abc? I think it works already.

**Comment by GideonBear** [comment]
> Are you importing abstractmethod from abc? I think it works already.

Not working both for `@abstractmethod` and `@property` + `@abstractmethod`

**Comment by mbyrnepr2** [comment]
I think its common-enough to have this pattern on the same line; but let's wait a bit for more feedback from maintainers.

**Comment by mbyrnepr2** [comment]
We could try a vote to make it easier:

1️⃣ for false-positive (we will remove the message when ellipsis is on the same line)
2️⃣ Do nothing (ellipsis should be on the following line)

**Comment by GideonBear** [comment]
It should be as easy as replacing [this](https://github.com/PyCQA/pylint/blob/7618b7e79ce64baa1eac0fd81950beb4b75318b8/pylint/checkers/format.py#L572-L574) with `return`

**Comment by mbyrnepr2** [comment]
I think you are good to go @GideonBear 😀
